tools/mason/MasonInit.chpl

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2004-2020 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use Path; 
+use Spawn;
+use MasonEnv;
+use MasonNew;
+use MasonBuild;
+use MasonHelp;
+use MasonUtils;
+use MasonExample;
+use FileSystem;
+
+proc masonInit(args) throws {
+	try! {
+		var name = '';
+		for arg in args[2..]{
+			if arg == '' {
+			name = arg;
+			} else {
+					name = arg;
+			}
+		}
+		//checks if path is given as parameter or not
+		if name == '' {
+			//If Mason.toml is already present, we get an error
+			if isFile("Mason.toml") {
+				throw new owned MasonError("Mason.toml already exists.");
+			} else {
+				//If Mason.toml not present, proceed with init
+				const cwd  = getEnv("PWD");
+				const currDir = basename(cwd);
+				var name = currDir;
+				var path = '.';
+				makeBasicToml(name,path);            
+			}
+		} else {
+			/*if a path is given as a parameter
+				check if the path is already present 
+				or if TOML file is already present 
+				in the given path
+			*/
+			var path = name;
+			if(isDir(path)) {
```

Comment:
I think this is reversed - we should throw an error if that directory does not already exist. The error could say something like:

```
"Directory does not exist: %s".format(path)
"Did you mean 'mason new' to create a new project from scratch?"
```

Reply:
Shouldn't mason check if a target directory has been already created and if not created, it should create one. Dlang's DUB does this.

Reply:
I believe this is the main difference of `mason init` vs. `mason new`:

- `mason init` creates a mason project in an existing directory.
- `mason new` creates a new mason project from scratch.

If a user tried `mason init <path>` on a _path_ that does not exist, they should be using `mason new` instead.

Reply:
Yeah. I understood that. That's why I deleted the previous comment 😜 
I'm working on it now. 👍